### ISSUE TITLE
gitconfig breaks commit info for detect command

### ISSUE THREAD

**Issue by martinhelfert** [gitleaks/gitleaks#1316]
**Describe the bug**
Running `gitleaks detect --source . -v -r gitleaks.json`  does not show commit specific infos. Fields like commit, message, author are all empty. The date field is empty or sometimes returns the value "0001-01-01T00:00:00Z"

**To Reproduce**

- I'm using delta to have a nicer view of diffs (using this config: https://github.com/dandavison/delta#get-started)
- Run `gitleaks detect --source . -v -r gitleaks.json` in a project where a key was commited
- The report output will look like this:

```json
[
 {
  "Description": "Generic API Key",
  "StartLine": 46,
  "EndLine": 47,
  "StartColumn": 18,
  "EndColumn": 1,
  "Match": "password=REDACTED",
  "Secret": "REDACTED",
  "File": "src/main/resources/application.properties",
  "SymlinkFile": "",
  "Commit": "",
  "Entropy": 3.75,
  "Author": "",
  "Email": "",
  "Date": "",
  "Message": "",
  "Tags": [],
  "RuleID": "generic-api-key",
  "Fingerprint": "src/main/resources/application.properties:generic-api-key:46"
 }
]
```

As a workaround, when settings $HOME to an empty string (so that the gitconfig is not applied), everything works as expected:

`HOME= gitleaks detect --source . -v -r ./gitleaks.json`

```json
[
 {
  "Description": "Generic API Key",
  "StartLine": 46,
  "EndLine": 47,
  "StartColumn": 18,
  "EndColumn": 1,
  "Match": "password=REDACTED",
  "Secret": "REDACTED",
  "File": "src/main/resources/application.properties",
  "SymlinkFile": "",
  "Commit": "",
  "Entropy": 3.75,
  "Author": "Firstname Lastname",
  "Email": "email@example.com",
  "Date": "2020-04-06T12:47:36Z",
  "Message": "commit message",
  "Tags": [],
  "RuleID": "generic-api-key",
  "Fingerprint": "src/main/resources/application.properties:generic-api-key:46"
 }
]
```

**Expected behavior**
Adjustments to git or gitconfig should not break gitleaks reports. The gitconfig should be ignored when running gitleaks or default settings should be applied when trying to determine the commit info.

**Basic Info (please complete the following information):**
 - OS: MacOS Sonoma 14.1.2 (23B92)
 - Gitleaks Version: 8.18.1

cc @zricethezav

**Comment by zricethezav** [comment]
Thanks for bringing this up @martinhelfert. I think there is a duplicate issue somewhere, let me see if I can dig that up.

**Comment by zricethezav** [comment]
@martinhelfert hmmm I'm unable to recreate this bug unfortunately. I was however able to find out a fix for https://github.com/gitleaks/gitleaks/issues/1206 which is a similar issue. 

> I'm using delta to have a nicer view of diffs (using this config: https://github.com/dandavison/delta#get-started)

I tried using this config and see the difference in my `git log -p` output but surprisingly gitleaks was still able to parse the output of `git log`. Were there any other values you changed in the gitconfig?

**Comment by martinhelfert** [comment]
> I tried using this config and see the difference in my `git log -p` output but surprisingly gitleaks was still able to parse the output of `git log`. Were there any other values you changed in the gitconfig?

Yes, you're correct. I did some testing and the settings that caused the issue with the commit info are these two:

```
[log]
  date = relative
```

and 

```
[format]
  pretty = format:%C(yellow)%h %Cblue%>(12)%ad %Cgreen%<(7)%aN%Cred%d %Creset%s
```

Whereby the use of either of the two settings already makes the commit info disappear.
